app/controllers/categories_controller.rb

```diff
@@ -0,0 +1,69 @@
+class CategoriesController < ApplicationController
+  before_action :set_category, only: %i[show edit update destroy]
+
+  # GET /categorys or /categorys.json
+  def index
+    @categories = Category.all
+    @category = Category.all.by_name
+  end
+
+  # GET /categorys/1 or /categorys/1.json
+  def show; end
+
+  # GET /categorys/new
+  def new
+    @category = Category.new
+  end
+
+  # GET /categorys/1/edit
+  def edit; end
+
+  # POST /categorys or /categorys.json
+  def create
+    @category = Category.new(category_params)
+
+    respond_to do |format|
+      if @category.save
+        format.html { redirect_to category_url(@category), notice: 'category was successfully created.' }
+        format.json { render :show, status: :created, location: @category }
+      else
+        format.html { render :new, status: :unprocessable_entity }
+        format.json { render json: @category.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # PATCH/PUT /categorys/1 or /categorys/1.json
+  def update
+    respond_to do |format|
+      if @category.update(category_params)
+        format.html { redirect_to category_url(@category), notice: 'category was successfully updated.' }
+        format.json { render :show, status: :ok, location: @category }
+      else
+        format.html { render :edit, status: :unprocessable_entity }
+        format.json { render json: @category.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # DELETE /categorys/1 or /categorys/1.json
+  def destroy
+    @category.destroy
+    respond_to do |format|
+      format.html { redirect_to categorys_url, notice: 'category was successfully destroyed.' }
```

Comment:
Any mention of `categorys` in this file should be `categories`. Most are comments, but this will fail.
```suggestion
      format.html { redirect_to categories_url, notice: 'category was successfully destroyed.' }
```